src/azul/chalice.py

```diff
@@ -107,7 +131,21 @@ def spec(self) -> JSON:
                 tag for tag in self._specs.get('tags', [])
                 if tag['name'] in used_tags
             ],
-            'servers': [{'url': self.self_url('/')}]
+            'servers': [{'url': self.self_url('/')}],
+            'security': [{self.app_name: self.oauth2_client.scopes}],
```

Comment:
Doesn't this indicate that all endpoints *require* authentication? If so, that would not be correct. I think this should declare the optionality but you should investigate:

```suggestion
            'security': [{}, {self.app_name: self.oauth2_client.scopes}],
```

Reply:
Also, `component` should precede `security` since the latter references the former. That's what all the examples look like.

Reply:
StackOverflow recapitulates your usage of the empty security schema and provides references to comments made by the OpenAPI developers.